### PR TITLE
Support nested EventBridge patterns, improve parity with AWS

### PR DESCRIPTION
**Pull request by whummer**
* support nested EventBridge patterns - addresses https://github.com/localstack/localstack/issues/5701
* start refactoring the integration tests, to reduce duplication - note: there is still an opportunity to do more unification, currently there are very similar tests duplicated for the different target types (SQS, SNS, Kinesis, EventBridge, ..)
* improve parity with AWS - two of the integration tests have already been executed and verified with `TEST_TARGET=AWS_CLOUD` - marked with `@pytest.mark.aws_validated`
* migrate tests to `pytest`
* Note: the change in the `events_starter.py`, in particular in `filter_event_based_on_event_format(..)` will affect the ASF migration in https://github.com/localstack/localstack/pull/5844 (will need to be migrated to that PR once merged). /cc @MEPalma

### CODE REVIEW

File: localstack/services/dynamodb/provider.py [hunk 1]
```diff
@@ -403,20 +404,21 @@ def create_table(
...
-        result["TableDescription"].pop("Tags", None)
         if tags:
-            table_arn = result["TableDescription"]["TableArn"]
+            table_arn = table_description["TableArn"]
             table_arn = self.fix_table_arn(table_arn)
             DynamoDBRegion.TABLE_TAGS[table_arn] = {tag["Key"]: tag["Value"] for tag in tags}
 
+        # remove invalid attributes from result
+        table_description.pop("Tags", None)
+        table_description.pop("BillingMode", None)
```
Comment (whummer) [inline localstack/services/dynamodb/provider.py:420]: Minor fix for ASF response serializer (removing superfluous response attribute), unrelated to the other changes in this PR

File: tests/integration/test_events.py [hunk 2]
```diff
@@ -104,168 +99,209 @@ def test_events_written_to_disk_are_timestamp_prefixed_for_chronological_orderin
...
+            events_client, sqs_client, pattern=TEST_EVENT_PATTERN, entries_asserts=[(entries, True)]
+        )
 
-        self.events_client.create_event_bus(Name=bus_name)
-        self.events_client.put_rule(
-            Name=rule_name,
-            EventBusName=bus_name,
-            EventPattern=json.dumps(TEST_EVENT_PATTERN),
+    @pytest.mark.aws_validated
+    def test_put_events_with_nested_event_pattern(self, events_client, sqs_client):
```
Comment (whummer) [inline tests/integration/test_events.py:145]: additional test cases for https://github.com/localstack/localstack/issues/5701

File: tests/integration/fixtures.py [hunk 3]
```diff
@@ -855,6 +856,10 @@ def factory(**kwargs) -> str:
 )
 
 
+# define markers programmatically via MarkGenerator (could also be moved to pytest.ini)
+MarkGenerator._markers.add("aws_validated")
```
Comment (dominikschubert) [inline tests/integration/fixtures.py:860]: We should probably put this into the "root" conftest.py for now :thinking:

File: localstack/services/events/events_starter.py [hunk 4]
```diff
@@ -65,14 +65,15 @@ def handle_numeric_conditions(conditions: List[Any], value: float):
     return True
 
 
-# TODO: refactor/simplify
+# TODO: refactor/simplify!
 def filter_event_based_on_event_format(self, rule_name: str, event: Dict[str, Any]):
     def filter_event(event_pattern_filter: Dict[str, Any], event: Dict[str, Any]):
         for key, value in event_pattern_filter.items():
-            event_value = event.get(key.lower())
+            event_value = event.get(key.lower(), event.get(key))
```
Comment (dominikschubert) [inline localstack/services/events/events_starter.py:72]: is the intent here to match keys in a case-agnostic way?

File: tests/integration/test_events.py [hunk 5]
```diff
@@ -104,168 +99,209 @@ def test_events_written_to_disk_are_timestamp_prefixed_for_chronological_orderin
...
-    def test_put_events_with_target_sqs(self):
-        queue_name = "queue-{}".format(short_uid())
-        rule_name = "rule-{}".format(short_uid())
-        target_id = "target-{}".format(short_uid())
-        bus_name = "bus-{}".format(short_uid())
-
-        sqs_client = aws_stack.create_external_boto_client("sqs")
-        queue_url = sqs_client.create_queue(QueueName=queue_name)["QueueUrl"]
-        queue_arn = aws_stack.sqs_queue_arn(queue_name)
+    @pytest.mark.aws_validated
```
Comment (dominikschubert) [inline tests/integration/test_events.py:131]: :partying_face:

**Review by dominikschubert — APPROVED** [review]
minor nits, otherwise LGTM :rocket: 

Awesome to see more and more AWS/LS agnostic (parity) tests :+1:

File: tests/integration/fixtures.py (hunk 3 above)
Comment (whummer) [reply tests/integration/fixtures.py:860]: Good point, updated 👍

File: localstack/services/events/events_starter.py (hunk 4 above)
Comment (whummer) [reply localstack/services/events/events_starter.py:73]: That's right. Added a comment to the code, to make this a bit more clear. 👍

**Pull request merged**